File: Graph_Representation/javascript/Adjacency_list.js
// JavaScript Implementation
// Adjacency List

class Graph {
    constructor(noOfVertices)
    {
        this.noOfVertices = noOfVertices;
        this.AdjList = new Map();
    }

    // Adding Nodes
    addVertex(v)
    {
        this.AdjList.set(v, []);
    }

    // Adding Edges
    addEdge(v, w)
    {

        this.AdjList.get(v).push(w);
        this.AdjList.get(w).push(v);
    }

    // Printing adjacency list
    printGraph()
    {
        var get_keys = this.AdjList.keys();
        for (var i of get_keys) 
        {
            var get_values = this.AdjList.get(i);
            var conc = "";
            for (var j of get_values)
                conc += j + " ";
            console.log(i + " -> " + conc);
        }
    }
}




var g = new Graph(6);
var vertices = [ '1', '2', '3', '4', '5', '6' ];

for (var i = 0; i < vertices.length; i++) {
    g.addVertex(vertices[i]);
}

g.addEdge('1', '3');
g.addEdge('1', '4');
g.addEdge('2', '4');
g.addEdge('3', '2');
g.addEdge('4', '5');
g.addEdge('5', '6');

g.printGraph();

File: Graph_Representation/java/Adjacency_list.java
// Adjascency List representation in Java

import java.util.*;

class Adjacency_list {

  // Adding Edges
  static void addEdge(ArrayList<ArrayList<Integer>> am, int s, int d) {
    am.get(s).add(d);
    am.get(d).add(s);
  }

  // Printing the graph
  static void printGraph(ArrayList<ArrayList<Integer>> am) {
    for (int i = 0; i < am.size(); i++) {
      System.out.println(i + ":");
      for (int j = 0; j < am.get(i).size(); j++) {
        System.out.print(" -> " + am.get(i).get(j));
      }
      System.out.println();
    }
  }

  public static void main(String[] args) {

    int V = 6;
    ArrayList<ArrayList<Integer>> am = new ArrayList<ArrayList<Integer>>(V);

    for (int i = 0; i < V; i++)
      am.add(new ArrayList<Integer>());


    addEdge(am, 0, 3);
    addEdge(am, 0, 2);
    addEdge(am, 1, 3);
    addEdge(am, 2, 1);
    addEdge(am, 3, 4);
    addEdge(am, 4, 5);


    System.out.println("\n");
    printGraph(am);
    System.out.println("\n");
  }


}